docs/bounding_volume_hierarchy.md

```diff
@@ -0,0 +1,187 @@
+```
+1) Default constructor.  Constructs an empty tree.  
+2) Constructs a bounding volume hierarchy from the given data source.
+## Parameters
+`primitives`
+: geometrical objects one wishes to index.
+#### Type requirements
+A specialization of [`ArborX::Traits::Access`](https://github.com/arborx/ArborX/blob/docs/docs/traits.md#arborxtraitsaccess) must match `Primitives` as the first template argument and `PrimitivesTag` as second argument.  
+The return type of `ArborX::Traits::Access<Primitives,PrimitivesTag>::get()` must decay either to `ArborX::Point` or `ArborX::Box`.
+ArborX provides specializations for Kokkos views but a user may specialize it for their types.
+
+## Complexity
+$O(N log(N))$, where `N = ArborX::Traits::Access<Primitives,PrimitivesTag>::size(primitives)` is the number of primitives passed to the constructor.
+## Exceptions
+Memory allocation with Kokkos may throw.
+## Notes
+
+## Example
+```C++
+#include <ArborX_LinearBVH.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <iostream>
+
+std::ostream &operator<<(std::ostream &os, ArborX::Point const &p)
+{
+  os << '(' << p[0] << ',' << p[1] << ',' << p[2] << ')';
+  return os;
+}
+
+int main(int argc, char *argv[])
+{
+  Kokkos::initialize(argc, argv);
+
+  Kokkos::View<ArborX::Point *> cloud("point_cloud", 1000);
+  Kokkos::parallel_for(1000, KOKKOS_LAMBDA(int i) {
+    cloud[i] = {{(float)i, (float)i, (float)i}};
+  });
+
+  using device_type = decltype(cloud)::device_type;
+  ArborX::BVH<device_type> bvh{cloud};
+
+  auto const box = bvh.bounds();
+  std::cout << box.minCorner() << " - " << box.maxCorner() << '\n';
+
+  Kokkos::finalize();
+
+  return 0;
+}
+```
+Output
+```
+(0,0,0) - (999,999,999)
+```
+## See also
+[`query`](#arborxbvhdevicetypequery)
+: search for all primitives that meet some predicates.  
+[`bounds`](#arborxbvhdevicetypebounds)
+: returns the bounding volume that contains all leaves.
+
+# `ArborX::BVH<DeviceType>::size()`
+```C++
+size_type size() const noexcept;
+```
+Returns the number of leaves in the tree.
+## Parameters
+(none)
+## Return value
+The number of leaves in the tree.
+## Complexity
+Constant.
+## See also
+[`empty`](#arborxbvhdevicetypeempty)
+: checks whether the tree is empty.  
+[`bounds`](#arborxbvhdevicetypebounds)
+: returns the bounding volume that contains all leaves.
+
+# `ArborX::BVH<DeviceType>::empty()`
+```C++
+bool empty() const noexcept;
+```
+Checks if the tree has no leaves.
+## Parameters
+(none)
+## Return value
+`true` if the tree is empty, `false` otherwise.
+## Complexity
+Constant.
+## See also
+[`size`](#arborxbvhdevicetypesize)
+: returns the number of leaves.
+
+# `ArborX::BVH<DeviceType>::bounds()`
+```C++
+bounding_volume_type bounds() const noexcept;
+```
+Returns the smallest axis-aligned box able to contain all leaves stored in the tree.
+## Parameters
+(none)
+## Return value
+The smallest bounding volume that contains all leaves stored in the tree or an invalid bounding volume if the tree is empty.
```

Comment:
```suggestion
The smallest bounding volume that contains all leaves stored in the tree. If the tree is empty, the return value is undefined.
```

Reply:
"Undefined" in C++ usually means unsafe.

Reply:
@masterleinad @Rombur Please break the tie.

Reply:
I like "invalid bounding volume" more.

Reply:
OK